sbncode/CAFMaker/FillReco.cxx

```diff
@@ -419,6 +419,54 @@ namespace caf
     }
   }
 
+  void FillSliceOpT0Finder(const std::vector<art::Ptr<sbn::OpT0Finder>> &opt0_v,
+                           caf::SRSlice &slice)
+  {
+    if (opt0_v.empty()==false){
+      unsigned int nopt0 = opt0_v.size();
+      double max_score=-1.; // score of the opt0 object with the highest score 
+      double sec_score=-1.; // score of the opt0 object with the 2nd highest score 
+
+      unsigned int max_idx = 0;
+      unsigned int sec_idx = 0;
+
+      // fill the default, which is the maximum 
+      for (unsigned int i = 0; i < nopt0; i++ ) {
+        const sbn::OpT0Finder &thisOpT0 = *opt0_v[i];
+        if (thisOpT0.score > max_score){
+          max_score = thisOpT0.score;
+          max_idx = i;
+        }
+      }
```

Comment:
Could do a `std::sort` function here just directly on the vector? Then use entries 0 and 1?

Reply:
this may be a stupid c++ question, but it's a `std::vector<sbn::OpT0Finder>` objects, rather than a vector of OpT0 scores... so I think that I can't directly sort the vector?

Reply:
i forgot you can give a function!! I will try this!

Reply:
Yeah exactly, something along the lines of 

`std::sort(opt0_v.begin(), opt0_v.end(),`
`          [](auto const &a, auto const &b)`
`          { return a->score > b->score; });`

You'll need to make the vector not `const` but then you can just query entry 0 and then entry 1 in the next if statement.